src/hotspot/share/services/memSnapshot.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_SERVICES_MEMSNAPSHOT_HPP
+#define SHARE_SERVICES_MEMSNAPSHOT_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+struct MemSnapshotPair {
+  size_t reserved;
+  size_t committed;
+};
+
+struct MemSnapshotOptions {
+  bool update_thread_stacks;
+  bool include_malloc;
+  bool include_vm;
+};
+
+class MemSnapshot : public CHeapObj<mtNMT> {
```

Comment:
I assume you don't think `NMTSnapshot` is good much better =) I can't come up with something much better than your suggestions, `NMTUsage` is not perfect either (sound like the usage cause by NMT). If we don't go with the NMT-prefix I think something like `MemUsage`or `NativeMemoryUsage` would make sense. Looking at the other reporting classes connected to NMT they are named `Mem`-something. So I think `MemUsage`, what do you think?

Reply:
Good. Up to you.

I wish we would allow C++ namespaces, having all in something like "namespace nmt". Then it does not hurt so much if names are a bit more generic.

Reply:
I'll go with `MemUsage`, normally I would have spent some more time trying to come up with something better, but Thursday is getting closer :)

Reply:
We always can revamp.